src/pages/index.js

```diff
@@ -4,22 +4,103 @@ import Section from "../components/Section.js";
 import PopupWithImage from "../components/PopupWithImage.js";
 import PopupWithForm from "../components/PopupWithForm.js";
 import UserInfo from "../components/UserInfo.js";
-import { initialCards, validationConfig } from "../utils/constants.js";
+import Api from "../components/Api.js";
+import PopupWithConfirm from "../components/PopupWithConfirm.js";
+import { validationConfig } from "../utils/constants.js";
 import "../pages/index.css";
 
+// API Initialization
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "81e44564-32cf-4c2a-b8fb-18b8f71d8ac2",
+    "Content-Type": "application/json",
+  },
+});
+
 // Instantiate PopupWithImage
 const imagePopup = new PopupWithImage("#image-modal");
 imagePopup.setEventListeners();
 
+// Delete Confirmation Popup
+const deleteConfirmPopup = new PopupWithConfirm(
+  "#delete-card-modal",
+  (cardId) => {
+    deleteConfirmPopup.renderLoading(true);
+    api
+      .deleteCard(cardId)
+      .then(() => {
+        const cardElement = document.querySelector(
+          `[data-card-id="${cardId}"]`
+        );
+        if (cardElement) {
+          cardElement.remove();
+        }
+        deleteConfirmPopup.close();
+      })
+      .catch((err) => console.error(err))
+      .finally(() => {
+        deleteConfirmPopup.renderLoading(false);
+      });
+  }
+);
+deleteConfirmPopup.setEventListeners();
+
+function handleDeleteButtonClick(cardId) {
+  deleteConfirmPopup.open(cardId);
+}
+
+function handleLikeButtonClick(card, cardId) {
+  const isLiked = card.isLiked();
+
+  const likeAction = isLiked ? api.dislikeCard(cardId) : api.likeCard(cardId);
+
+  likeAction
+    .then((updatedCard) => {
+      card.setLikes(updatedCard.isLiked);
+    })
+    .catch((err) => {
+      console.error("Like operation failed:", err);
+
+      if (card._likeButton) {
+        card._likeButton.disabled = false;
```

Comment:
`_likeButton`

Please, delete the **low dash** (`underscore`) from methods/fields if you want to use them **outside** a class because the `low dash` shows a **private** field which is used only inside the class.

Reply:
My mistake! All fixed!